apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
Not 100% sure when we should emit ```saved``` - from one side it should be called before ```store``` to be consistent with ```delete``` from the other side it's actually saved when store is finished. What do you think?